test/parallel/test-quic-internal-setcallbacks.js

```diff
@@ -0,0 +1,39 @@
+// Flags: --expose-internals
+'use strict';
+
+require('../common');
+const { internalBinding } = require('internal/test/binding');
+const quic = internalBinding('quic');
+
+const { throws } = require('assert');
+
+const callbacks = {};
+
+const fns = [
+  'onEndpointClose',
+  'onSessionNew',
+  'onSessionClose',
+  'onSessionDatagram',
+  'onSessionDatagramStatus',
+  'onSessionHandshake',
+  'onSessionPathValidation',
+  'onSessionTicket',
+  'onSessionVersionNegotiation',
+  'onStreamCreated',
+  'onStreamBlocked',
+  'onStreamClose',
+  'onStreamReset',
+  'onStreamHeaders',
+  'onStreamTrailers',
+];
+
+// Each iteration should fail because of a missing callback.
+for (const fn of fns) {
+  throws(() => quic.setCallbacks(callbacks), {
+    code: 'ERR_MISSING_ARGS',
+  });
+  callbacks[fn] = () => {};
+}
+
+// By this point we've set all the callbacks, so it should work
+quic.setCallbacks(callbacks);
```

Comment:
Might be better to do something like this to check if any single callback is missing from the list it will fail. As the test is right now, it's possible that only the _last_ callback in the list is actually required and the others will just fail because it hasn't got to that one yet.

```suggestion
const callbacks = {
  onEndpointClose() {},
  onSessionNew() {},
  onSessionClose() {},
  onSessionDatagram() {},
  onSessionDatagramStatus() {},
  onSessionHandshake() {},
  onSessionPathValidation() {},
  onSessionTicket() {},
  onSessionVersionNegotiation() {},
  onStreamCreated() {},
  onStreamBlocked() {},
  onStreamClose() {},
  onStreamReset() {},
  onStreamHeaders() {},
  onStreamTrailers() {},
};

// Fail if any callback is missing
for (const fn of Object.keys(callbacks)) {
  const { [fn]: _, ...rest } = callbacks;
  throws(() => quic.setCallbacks(rest), {
    code: 'ERR_MISSING_ARGS',
  });
}

// If all callbacks are present it should work
quic.setCallbacks(callbacks);
```